marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java

```diff
@@ -1700,20 +1766,105 @@ public interface ModifyPlan extends PreparePlan, PlanBuilderBase.ModifyPlanBase
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan joinDocCols(PlanDocColsIdentifier docCols, String docIdCol);
+/**
+  * Populates view with uri, doc, collections, metadata, permissions, and/or quality document descriptor columns for database document values.
+  * @param docCols  Supplies a document column identifier object to specify which document columns to join and their names. The names cannot conflict with existing column names.
+  * @param docIdCol  Identifies the document. See {@link PlanBuilder#col(XsStringVal)}
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan joinDocCols(PlanDocColsIdentifier docCols, PlanColumn docIdCol);
+/**
+  * Validates a column that has in-memory nodes as values against a schema.
+  * @param validated  Identifies a document content column. See {@link PlanBuilder#col(XsStringVal)}
+  * @param schemaDef  Defines the schema using keys - kind, schemaUri and mode.
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan validateDoc(String validated, PlanSchemaDef schemaDef);
+/**
+  * Validates a column that has in-memory nodes as values against a schema.
+  * @param validated  Identifies a document content column. See {@link PlanBuilder#col(XsStringVal)}
+  * @param schemaDef  Defines the schema using keys - kind, schemaUri and mode.
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan validateDoc(PlanColumn validated, PlanSchemaDef schemaDef);
+/**
+  * Creates sequence valued columns in SQL and SPARQL.
+  * @param inputColumn  Supplies the input. If the value is NULL the unnest-inner has no output. See {@link PlanBuilder#col(XsStringVal)}
+  * @param valueColumn  Specifies the output column. If the valueColumn is specified we use that as output, else we use 'unnest' as output. See {@link PlanBuilder#col(XsStringVal)}
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan unnestInner(String inputColumn, String valueColumn);
+/**
+  * Creates sequence valued columns in SQL and SPARQL.
+  * @param inputColumn  Supplies the input. If the value is NULL the unnest-inner has no output. See {@link PlanBuilder#col(XsStringVal)}
+  * @param valueColumn  Specifies the output column. If the valueColumn is specified we use that as output, else we use 'unnest' as output. See {@link PlanBuilder#col(XsStringVal)}
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan unnestInner(PlanExprCol inputColumn, PlanExprCol valueColumn);
+/**
+  * Creates sequence valued columns in SQL and SPARQL.
+  * @param inputColumn  Supplies the input. If the value is NULL the unnest-inner has no output. See {@link PlanBuilder#col(XsStringVal)}
+  * @param valueColumn  Specifies the output column. If the valueColumn is specified we use that as output, else we use 'unnest' as output. See {@link PlanBuilder#col(XsStringVal)}
+  * @param ordinalColumn   See {@link PlanBuilder#col(XsStringVal)}
```

Comment:
@ehennum I could not find a good definition for "ordinalColumn" here - https://wiki.marklogic.com/display/ENGINEERING/Optic+UNNEST+Support . Also, rest of the explanation is based on the info i could find on this wiki page, is there a different functional spec I should be looking at?